src/main/java/core/basesyntax/service/PurchaseOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Inventory;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements FruitOperationHandler {
+    @Override
+    public void getFruitOperation(FruitTransaction transaction, Inventory inventory) {
+        inventory.getInventory().put(transaction.getFruit(),
+                inventory.getInventory().getOrDefault(
+                        transaction.getFruit(), 0) - transaction.getQuantity());
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Right now, your method deducts the quantity without checking if the resulting balance would be positive, which could lead to negative stock quantities.